### DISCUSSION TITLE
[Docker][Linux] Why so slow?

### DISCUSSION THREAD

**Discussion by mike7800** [cypress-io/cypress#9137] [Docker]
I'm running Cypress with the command:  
 `docker run -it --rm -e "DEBUG=cypress:*" --volume $PWD/code:/e2e --workdir=/e2e --entrypoint=cypress cypress/included:5.5.0 run`
The test is very simple but it takes 58 seconds to complete. 
The test:
```
//simple_spec.js
describe('Test', () => {
  it('Visits Google', () => {
    cy.visit('http://www.google.com')
    cy.title().should('eq', 'Google')
  })
})
```

Here is the log https://pastebin.com/0jd7FuBc
Screenshot from the video
![cypress-video](https://user-images.githubusercontent.com/74206310/98581378-b4a9ee00-22c1-11eb-90e6-f298e688681c.png)

cypress.json is {}, so default configuration.
i have tried to run it with `--browser chrome` or `--browser firefox`; However, it also takes about 58 seconds. Removing `DEBUG` env or setting record `video` to `false` didn't change anything.

The same situation is when I try to use:  
 `docker run -it --rm --shm-size=8g --network host -v /var/run/dbus/system_bus_socket:/var/run/dbus/system_bus_socket --privileged -e DISPLAY -e "DEBUG=cypress:*" -v /tmp/.X11-unix:/tmp/.X11-unix -u node --volume $PWD/code:/e2e --workdir=/e2e --entrypoint=cypress cypress/included:5.5.0 open --project . --config baseUrl=http://localhost:4000`
![cypress-open](https://user-images.githubusercontent.com/74206310/98581526-e622b980-22c1-11eb-8fbb-5e32356a13ab.png)

If I do not close the window and press `Run all tests` button, then the second run is faster.
![cypress-open-rerun](https://user-images.githubusercontent.com/74206310/98581547-ee7af480-22c1-11eb-8cf0-f7ca49332194.png)

I was able to shorten time by setting `dns` param `docker run -it --rm --dns=8.8.8.8 -e "DEBUG=cypress:*" --volume $PWD/code:/e2e --workdir=/e2e --entrypoint=cypress cypress/included:5.5.0 run`

```
(Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  simple_spec.js                           00:18        1        1        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        00:18        1        1        -        -        -  

  cypress:server:cypress about to exit with code 0 +26s
  cypress:cli child event fired { event: 'exit', code: 0, signal: null } +29s
  cypress:cli Stopping Xvfb +29s
  cypress:cli child event fired { event: 'close', code: 0, signal: null } +4ms
```

On the native browser the page loads under one second. So it's still far from acceptable state.

I didn't find a solution for this problem :(. Anyone encountered such a problem? Is this some kind of bug or regression?

**Comment by diggabyte** [comment]
Docker can't leverage GPU acceleration (except in very specific cases, using a linux host w/ nvidia GPU and special docker images.. it's a mess). Only chromium supports gpu on linux and it's disabled by default by cypress, so you have to enable the gpu in browser launch flags. Search the cypress issues for GPU and you'll see it's not quite an easy thing to achieve.

I'd love to know if you have found any other ways to improve the performance in docker.
